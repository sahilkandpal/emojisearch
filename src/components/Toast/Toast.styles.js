import Styled from "styled-components";
export const StyledToast = Styled.div`
    position: absolute;
    top: 0;
    right: 50px;
    padding: 0px 50px;
    background-color: #000;
    color: #fff;
    transform: ${(props) =>
      props.isVisible ? "translateY(100%)" : "translateY(-100%)"};
    transition: 0.5s;
`;
