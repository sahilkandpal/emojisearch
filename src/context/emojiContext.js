import { createContext, useState, useContext } from "react";
import emojidata from "../json/emoji-en-US.json";
export const emojiContext = createContext({
  emojiData: {},
  filterEmojiData: () => {},
  handleEmojiData: () => {},
  loading: false,
  handleLoading: () => {},
});
const initialState = emojidata;

export const EmojiContextProvider = ({ children }) => {
  const [emojiData, setEmojiData] = useState(initialState);
  const [loading, setLoading] = useState(false);

  const handleLoading = (state) => {
    setLoading(state);
  };

  const handleEmojiData = (val) => {
    setEmojiData(val);
  };

  const filterEmojiData = (query) => {
    if (query.trim() !== "") {
      let newdata = {};
      Object.keys(emojidata).map((val) => {
        const filtered = emojidata[val].filter((value) =>
          value.toLowerCase().includes(query.trim().toLowerCase())
        );
        if (filtered.length > 0) {
          newdata = { ...newdata, [val]: emojidata[val] };
        }
        return null;
      });
      return newdata;
    } else {
      return emojidata;
    }
  };

  return (
    <emojiContext.Provider
      value={{
        emojiData,
        filterEmojiData,
        handleEmojiData,
        loading,
        handleLoading,
      }}
    >
      {children}
    </emojiContext.Provider>
  );
};

export const useEmojiContext = () => {
  const {
    emojiData,
    filterEmojiData,
    handleEmojiData,
    loading,
    handleLoading,
  } = useContext(emojiContext);
  return {
    emojiData,
    filterEmojiData,
    handleEmojiData,
    loading,
    handleLoading,
  };
};
