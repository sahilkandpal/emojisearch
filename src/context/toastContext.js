import { createContext, useState, useContext } from "react";

const initialState = { message: "", isVisible: false };

export const toastContext = createContext({
  toast: initialState,
  handleToast: () => {},
});

export const ToastContextProvider = ({ children }) => {
  const [toast, setToast] = useState(initialState);

  const handleToast = (message) => {
    setToast({ message, isVisible: true });
    setTimeout(() => {
      setToast({ message, isVisible: false });
    }, 1000);
  };

  return (
    <toastContext.Provider value={{ toast, handleToast }}>
      {children}
    </toastContext.Provider>
  );
};

export const useToastContext = () => {
  const { toast, handleToast } = useContext(toastContext);
  return {
    toast,
    handleToast,
  };
};
