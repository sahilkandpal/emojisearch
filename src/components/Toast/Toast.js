import React from "react";
import { useToastContext } from "../../context/toastContext";
import { StyledToast } from "./Toast.styles";
const Toast = () => {
  const { toast } = useToastContext();
  return (
    <StyledToast isVisible={toast.isVisible}>
      <h4>{toast.message}</h4>
    </StyledToast>
  );
};

export default Toast;
