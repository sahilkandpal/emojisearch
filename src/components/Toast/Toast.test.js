import React from "react";
import { shallow } from "enzyme";
import Toast from "./Toast";
import { useToastContext } from "../../context/toastContext";

jest.mock("../../context/toastContext");

describe("Given Input", () => {
  beforeEach(() => {
    useToastContext.mockReturnValue({
      toast: { message: "", isVisible: false },
    });
  });

  test("Toast should visible on isVisible true", () => {
    useToastContext.mockReturnValue({
      toast: { message: "ðŸ˜¢ copied!", isVisible: true },
    });
    const component = shallow(<Toast />);
    expect(component.props().isVisible).toEqual(true);
    expect(component).toMatchSnapshot();
  });

  test("Toast shouldn't be visible on isVisible false", () => {
    useToastContext.mockReturnValue({
      toast: { message: "", isVisible: false },
    });
    const component = shallow(<Toast />);
    expect(component.props().isVisible).toEqual(false);
    expect(component).toMatchSnapshot();
  });
});
