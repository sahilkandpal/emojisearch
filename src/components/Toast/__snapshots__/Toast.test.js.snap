// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Input Toast should visible on isVisible true 1`] = `
<styled.div
  isVisible={true}
>
  <h4>
    ðŸ˜¢ copied!
  </h4>
</styled.div>
`;

exports[`Given Input Toast shouldn't be visible on isVisible false 1`] = `
<styled.div
  isVisible={false}
>
  <h4 />
</styled.div>
`;
